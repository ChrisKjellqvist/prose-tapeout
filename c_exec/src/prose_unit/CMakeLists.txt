cmake_minimum_required(VERSION 3.5)
project(prose_unit_tests)
cmake_policy(SET CMP0074 NEW)

set(CMAKE_CXX_STANDARD 20)

#set(BEETHOVEN_ROOT "/Users/entropy/Development/Composer-Software/beethoven_dir" CACHE PATH "Path to Beethoven library")
option(USE_TORCH "Enable PyTorch testing" ON)
if (${USE_TORCH})
    message("Enabling PyTorch")
    find_package(Torch REQUIRED)
    add_compile_definitions(USE_TORCH=1)
    include_directories(${TORCH_INCLUDE_DIRS})
endif ()
find_package(beethoven REQUIRED)

add_subdirectory(unit)
add_subdirectory(golden)
add_subdirectory(utility)
add_subdirectory(hw_impl)
add_subdirectory(neo_test)
