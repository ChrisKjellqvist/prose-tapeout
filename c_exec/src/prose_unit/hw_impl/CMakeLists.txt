beethoven_library(
        prose_hw_impl
        SOURCES
            src/prose_impl.cc
            src/float_wrapper.cc)

target_include_directories(prose_hw_impl PUBLIC include)
target_link_libraries(prose_hw_impl PUBLIC prose_utils)

