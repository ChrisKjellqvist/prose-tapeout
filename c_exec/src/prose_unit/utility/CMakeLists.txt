add_library(
        prose_utils SHARED
        util.cc
        interchange_format.cc
        torch_util.cc)

# including TORCH_INCLUDE_DIRS should be harmless
target_include_directories(prose_utils PUBLIC ./ ${TORCH_INCLUDE_DIRS})
target_link_libraries(prose_utils PUBLIC ${TORCH_LIBRARIES})
beethoven_build(kill_sim SOURCES kill_sim.cc)