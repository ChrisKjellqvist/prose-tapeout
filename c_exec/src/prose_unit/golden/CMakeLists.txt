add_library(golden SHARED
        src/prose_golden.cc)
target_include_directories(golden PUBLIC ./include ${TORCH_INCLUDE_DIRS})
message(STATUS "TORCH_INCLUDE_DIRS: ${TORCH_INCLUDE_DIRS}")
target_link_libraries(golden PUBLIC ${TORCH_LIBRARIES} prose_utils)
